centreon-dsm/www/modules/centreon-dsm/core/configuration/services/formSlot.php

```diff
@@ -38,198 +38,212 @@
+        $form->addElement(
+            "button",
+            "change",
+            _("Modify"),
+            array(
+                "class" => "btc bt_default",
+                "onClick" => "javascript:window.location.href='?p=" . $p . "&o=c&pool_id=" . $pool_id . "'"
+            )
+        );
+        $form->setDefaults($pool);
+        $form->freeze();
+    } elseif ($o == "c") {
+        // Modify a pool information
+        $subC = $form->addElement('submit', 'submitC', _("Save"), array("class" => "btc bt_success"));
+        $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
+        $form->setDefaults($pool);
+    } elseif ($o == "a") {
+        // Add a pool information
+        $subA = $form->addElement('submit', 'submitA', _("Save"), array("class" => "btc bt_success"));
+        $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
+    }
 
-$valid = false;
-$msgErr = "";
-if ($form->validate() && $from_list_menu == false) {
-    $poolObj = $form->getElement('pool_id');
-    if ($form->getSubmitValue("submitA")) {
-        try {
-            $pId = insertpoolInDB();
-            $valid = true;
-            $poolObj->setValue($pId);
-        } catch (Exception $e) {
-            $valid = false;
-            $msgErr = $e->getMessage();
-        }
-    } elseif ($form->getSubmitValue("submitC")) {
-        try {
-            $valid = updatePoolInDB($poolObj->getValue());
-        } catch (Exception $e) {
-            $valid = false;
-            $msgErr = $e->getMessage();
+
+    if ($form->validate() && $from_list_menu == false) {
+        $poolObj = $form->getElement('pool_id');
+        if ($form->getSubmitValue("submitA")) {
+            try {
+                $pId = insertpoolInDB();
+                $valid = true;
+                $poolObj->setValue($pId);
+            } catch (Exception $e) {
+                $valid = false;
+                $msgErr = $e->getMessage();
+            }
+        } elseif ($form->getSubmitValue("submitC")) {
+            try {
+                $valid = updatePoolInDB($poolObj->getValue());
+            } catch (Exception $e) {
+                $valid = false;
+                $msgErr = $e->getMessage();
+            }
         }
+        $o = null;
+        $form->addElement(
+            "button",
+            "change",
+            _("Modify"),
+            array(
+                "class" => "btc bt_default",
+                "onClick" => "javascript:window.location.href='?p=" . $p . "&o=c&pool_id=" . $poolObj->getValue() . "'"
+            )
+        );
+        $form->freeze();
     }
-    $o = null;
-    $form->addElement(
-        "button",
-        "change",
-        _("Modify"),
-        array(
-            "class" => "btc bt_default",
-            "onClick" => "javascript:window.location.href='?p=" . $p . "&o=c&pool_id=" . $poolObj->getValue() . "'"
-        )
+}catch(Throwable $e){
```

Comment:
Your code style is not good, you need to use spaces ;)

Reply:
done

Reply:
good